cs3/storage/provider/v1beta1/resources.proto

```diff
@@ -283,6 +283,11 @@ message ResourceId {
   // uniquely identity the resource in the internal
   // implementation of the service.
   string opaque_id = 2;
+  // REQUIRED.
```

Comment:
I would argue that adding a new value that is `REQUIRED` IS indeed a breaking change. Also it will need some integration in ocis. The reason is:

We used `StorageID` (REQUIRED FIELD) as the spaceID before. After we added the StorageProviderID (OPTIONAL FIELD) we used this field for both so we had `StorageProviderID$SpaceID` or only `SpaceID`

Now that would be completely different: `StorageID` would now be the StorageProviderID or empty. While the new SpaceID field would be REQUIRED.

Reply:
Additionally I doubt that it would be compatible with edge without changes. Did you test that?

Reply:
It is only "REQUIRED" on the edge branch. The master branch will not need to change.

> Now that would be completely different: StorageID would now be the StorageProviderID or empty. While the new SpaceID field would be REQUIRED.

Yes, exactly. That means a clean implementation of spaces in reva edge and ocis.

Reply:
The `REQUIRED` comment is not enforced. that is why this change will work as is in both: `master` and `edge`. In `edge` we can then get rid of the storageid splitting hack and continue with a clean api.

Reply:
I agree with @kobergj that this IS a breaking change. Rather, I propose to leave the `storage_id` as `REQUIRED` and add the new `space_id` as `OPTIONAL`.

IMHO adding this field is a welcome extension we (I, at least) advocated, to allow having a cleaner and more self-explanatory code in Reva edge, but at the same time I'd argue that the whole concept of spaces, at least at the API level, is optional, as we can have implementations (Reva master) that are doing without them. That's why the `space_id` should be `OPTIONAL`.

Reply:
Ok, let us rephrase it that it is a **needed** change. whether it is breaking or not. I meant not breaking because `master` needs no code changes.

Reply:
Honestly with these `REQUIRED` and `OPTIONAL` values they are a bit pointless.

on the `master` branch `space_id` is optional
on the `edge` branch `storage_id` is optional

Reply:
spaces are the future so we should make the `storage_id` optional.

I can also live with it vice versa. These flags have no effect on the code.